File: DynamoDB-SDK-Examples/node.js/WorkingWithItems/delete-item.js
/* This is an example of a DeleteCommand with a ConditionExpression using
the higher level DocumentClient for Amazon DynamoDB. The conditional
expression must be true for this DeleteCommand to succeed. */

const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { DynamoDBDocumentClient, DeleteCommand } = require("@aws-sdk/lib-dynamodb");

async function deleteItem() {
  const client = new DynamoDBClient({ region: "us-west-2" });
  const ddbDocClient = DynamoDBDocumentClient.from(client);
  return await ddbDocClient.send(
      new DeleteCommand({
        TableName: "RetailDatabase",
        Key: {
          pk: "jim.bob@somewhere.com",    // Partition key
          sk: "metadata",                 // Sort key
        },
        // Below is an optional conditional expression to validate the value of the address.pcode
        // value is the value provided. If the item matching the partition and sort key are found
        // but the values below does not match, then the delete will fail and the item not deleted.
        ConditionExpression: "address.pcode = :val",
        ExpressionAttributeValues: {
          ":val" : "98261",
        },
      })
  );
}

deleteItem()
    .then((data) =>
        console.log("DeleteCommand succeeded:", JSON.stringify(data, null, 2)))
    .catch((error) => console.error(JSON.stringify(error, null, 2)));

File: examples/SDK/kotlin/data_plane/WorkingWithItems/src/main/kotlin/DeleteItem.kt
import aws.sdk.kotlin.services.dynamodb.DynamoDbClient
import aws.sdk.kotlin.services.dynamodb.deleteItem
import aws.sdk.kotlin.services.dynamodb.model.AttributeValue

suspend fun deleteItem(
    tableName: String,
    key: Map<String, AttributeValue>,
) {
    DynamoDbClient { region = "us-west-2" }.use { ddb ->
        ddb.deleteItem {
            this.tableName = tableName
            this.key = key
        }
        println("Deleted item with the given key.")
    }
}

suspend fun main() {
    val tableName = "YourTableName"
    val key = mapOf("PrimaryKey" to AttributeValue.S("YourPrimaryKeyValue"))

    deleteItem(tableName, key)
}
